gdsfactory/geometry/fillet.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import annotations
+
+from typing import Union, List
+
+import gdstk
+
+from gdsfactory.typings import Component, ComponentReference
+
+
+def fillet(
+    operand: Union[ComponentReference, Component, gdstk.Polygon],
+    radius: Union[float, List[float]],
+    tolerance: float = 0.01,
+) -> List[gdstk.Polygon]:
+    """Perform a fillet operation and return the list of resulting Polygons.
+
+    Args:
+        operand: polygon, Component, .
+        radius, (number or sequence) – Fillet radius.
+            A sequence of values can be used to define the fillet radius for each vertex.
+        tolerance: Tolerance used for calculating the polygonal approximation of the filleted corners.
+
+    """
+
+    if hasattr(operand, "get_polygons"):
+        operand = operand.get_polygons(as_array=False)
```

Comment:
For a `Component`, I would expect `fillet` to not only modify the geometry contained directly by that component, but also any geometry contained by subcomponents.

Reply:
yes, get_polygons should give you the subcomponent polygons